python/spark_sklearn/keyed_models.py

```diff
@@ -328,7 +398,10 @@ def fitEstimator(_, pandasDF):
             # Potential optimization - del pandasDF
 
             estimatorClone = sklearn.base.clone(estimator)
-            estimatorClone.fit(X, y)
+            if y is None:
```

Comment:
I don't think so - if for some reason you had an empty array then not y would be true.